### DISCUSSION TITLE
Group tests related to customisation

### DISCUSSION THREAD

**Discussion by Quentinchampenois** [OpenSourcePolitics/osp-app-k8s#69] [Ideas]
What are conventions when adding tests related to application customisation ? 

In fact, a decidim-app can have some unique customisations that we should test as well. For example, at the moment we implement the decidim module `term-customizer`, and we might add another module in the future. So it could be interesting to add tests to ensure the modules are working correctly and does not cause unexpected conflicts.

So, I was thinking about solutions like :

- Adding a new github action workflow for customisations, named `Customisation tests` for example.
- Defining a pattern name convention for specs files like :
    - `with_term_customizer_spec.rb`
    - `with_decidim_awesome_spec.rb`
    - `with_[module_name]_spec.rb`

And those files could be located in `spec/system/customisations/`

What do you think about this solution, do you have any other suggestion ?

**Comment by armandfardeau** [comment]
I disagree. We should test for the final rendering on this particular application. We should not test every special case, as there is no case. If a gem is added, the tests must be updated.

A test suite may be different from an instance to another.

**Reply by Quentinchampenois** [reply]
Thanks for the feedback ! 
Ok I understand and I agree with you
